src/containers/my-cooperations/add-review-modal/AddReviewModal.tsx

```diff
@@ -0,0 +1,141 @@
+import { useTranslation } from 'react-i18next'
+
+import Box from '@mui/material/Box'
+import Typography from '@mui/material/Typography'
+import Rating from '@mui/material/Rating'
+import TextField from '@mui/material/TextField'
+
+import Button from '~/design-system/components/button/Button'
+
+import { ReviewService } from '~/services/review-service'
+import { useAppSelector, useAppDispatch } from '~/hooks/use-redux'
+import useForm from '~/hooks/use-form'
+import { openAlert } from '~/redux/features/snackbarSlice'
+import { snackbarVariants } from '~/constants'
+import { getErrorKey } from '~/utils/get-error-key'
+import { useModalContext } from '~/context/modal-context'
+import {
+  initialValues,
+  validations
+} from '~/containers/my-cooperations/add-review-modal/AddReviewModal.constants'
+
+import {
+  ComponentEnum,
+  ReviewDataFromCooperation,
+  ReviewData,
+  ErrorResponse,
+  DataFromCooperation
+} from '~/types'
+import { styles } from '~/containers/my-cooperations/add-review-modal/AddReviewModal.styles'
+import useMutation from '~/hooks/use-mutation'
+
+interface AddReviewModalProps {
+  data: ReviewDataFromCooperation
+}
+
+const AddReviewModal: React.FC<AddReviewModalProps> = ({ data }) => {
+  const { t } = useTranslation()
+  const { userRole } = useAppSelector((state) => state.appMain)
+  const dispatch = useAppDispatch()
+  const { closeModal } = useModalContext()
+
+  const handleResponse = () => {
+    dispatch(
+      openAlert({
+        severity: snackbarVariants.success,
+        message: t('cooperationsPage.cooperationDetails.success')
+      })
+    )
+    closeModal()
+  }
+
+  const handleResponseError = (error?: ErrorResponse) => {
+    dispatch(
+      openAlert({
+        severity: snackbarVariants.error,
+        message: getErrorKey(error)
+      })
+    )
+  }
+
+  const { mutate: submitReview } = useMutation({
+    mutationFn: (data: {
+      dataFromCooperation: DataFromCooperation
+      comment: string
+      rating: number
+    }) => {
+      return ReviewService.submitReview({
+        ...data.dataFromCooperation,
+        comment: data.comment,
+        rating: data.rating
+      })
+    },
+    onSuccess: handleResponse,
+    onError: handleResponseError
+  })
+
+  const {
+    data: reviewData,
+    handleInputChange,
+    handleNonInputValueChange,
+    handleSubmit
+  } = useForm<ReviewData>({
+    initialValues,
+    validations,
+    onSubmit: (reviewData) => {
+      if (reviewData) {
+        submitReview({
+          dataFromCooperation: data,
```

Comment:
if you will do what I suggested with `mutationFn`, you will have to do this ↓

```suggestion
          ...data,
```